### PR TITLE
Feature/improved turnsettings

### PR DESCRIPTION
**Pull request by Romeeey**
# To test
* [ ] Open Settings menu and check if turn speed slider is there
* [ ] Change slider and see how turning gets slower and faster
* [ ] Check for snapturn and continuous turn
* [ ] Check if items push you back when you move them to your body center (It should not)
* [ ] Walk behind the menu (it should not be invisible anymore, although flipped)

# Checklist
(Please judge for yourself which entry is valuable for your PR)

## Scene
* [ ] FeatureFlags reverted to right value

# Testing
* [x] Merged _main_ into this branch and tested with the latest features
* [x] Tested with PCVR
* [ ] Tested with Pico / Quest

## Dependencies
* [ ] If libphoenix-shared.dll is changed, .so is also changed

### CODE REVIEW

**Review by JucanAndreiDaniel — CHANGES_REQUESTED** [review]
Bootstrap.unity seem to still have merge issues, please fix this before i can test

File: Assets/GothicVR/Resources/Materials/BackgroundMaterial.mat.rej
Comment (JaXt0r) [inline Assets/GothicVR/Resources/Materials/BackgroundMaterial.mat.rej:1]: Is this file needed? It's content look like it's some leftover from a merge?

File: Assets/GothicVR/Scenes/Bootstrap.unity [hunk 1]
```diff
@@ -1064,6 +1064,10 @@ MonoBehaviour:
   VObPerFrame: 75
   moveSpeedPlayerPref: MoveSpeed
   turnSettingPlayerPref: TurnSetting
+<<<<<<< HEAD
```
Comment (JaXt0r) [inline Assets/GothicVR/Scenes/Bootstrap.unity:1067]: Merge issues. Please solve. From what I see, I think the first one is right. The other doesn't contain any value.

File: Assets/GothicVR/Scripts/Manager/ConstantsManager.cs [hunk 2]
```diff
@@ -19,6 +19,7 @@ public class ConstantsManager : SingletonBehaviour<ConstantsManager>
         //Collection of PlayerPref entries for settings
         public string moveSpeedPlayerPref = "MoveSpeed";
         public string turnSettingPlayerPref = "TurnSetting";
+        public string turnAngelPlayerPref = "TurnAngelSetting";
```
Comment (JaXt0r) [inline Assets/GothicVR/Scripts/Manager/ConstantsManager.cs:22]:
```suggestion
        public string turnAnglePlayerPref = "TurnAngleSetting";
```

Angel == 🎣 in German
Angle == 📐in English
;-)

File: Assets/GothicVR/Scripts/UI/TurnAngelController.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.XR.Interaction.Toolkit;
+using GVR.Manager;
+
+public class TurnAngelController : MonoBehaviour
```
Comment (JaXt0r) [inline Assets/GothicVR/Scripts/UI/TurnAngelController.cs:6]:
```suggestion
public class TurnAngleController : MonoBehaviour
```

**Review by JaXt0r — CHANGES_REQUESTED** [review]
Just a few question and name improvements. I will test it after changes are done.

File: Assets/GothicVR/Scenes/General.unity [hunk 4]
```diff
@@ -1458,86 +1454,6 @@ PrefabInstance:
...
-      value: Position
-      objectReference: {fileID: 0}
-    - target: {fileID: 3286136463054414637, guid: 4600f492db92202478d7c53019a519d0, type: 3}
-      propertyPath: m_RotationInput.m_Action.m_SingletonActionBindings.Array.data[0].m_Action
-      value: Rotation
-      objectReference: {fileID: 0}
-    - target: {fileID: 3286136463054414637, guid: 4600f492db92202478d7c53019a519d0, type: 3}
-      propertyPath: m_RotationInput.m_Action.m_SingletonActionBindings.Array.data[1].m_Action
-      value: Rotation
-      objectReference: {fileID: 0}
```
[the remark this answers is not in the document]
Comment (Romeeey) [reply Assets/GothicVR/Scenes/General.unity:1540]: No. I'll revert.

File: Assets/GothicVR/Resources/Materials/BackgroundMaterial.mat.rej
Comment (Romeeey) [reply Assets/GothicVR/Resources/Materials/BackgroundMaterial.mat.rej:1]: not needed.

File: Assets/GothicVR/Scripts/UI/TurnAngelController.cs [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+{
+    public ActionBasedSnapTurnProvider snapturnprovider;
+    public ActionBasedContinuousTurnProvider continuousturnprovider;
+    void Start()
+    {
+        Slider turnangelslider = transform.GetComponent<Slider>();
+        turnangelslider.onValueChanged.AddListener(ChangeTurnAngel);
+        turnangelslider.value = PlayerPrefs.GetFloat(ConstantsManager.I.turnAngelPlayerPref, 45);
```
[the remark this answers is not in the document]
Comment (Romeeey) [reply Assets/GothicVR/Scripts/UI/TurnAngelController.cs]: Followed your suggestions. Movespeed and turnangel can now be configured in the constants manager.

File: Assets/GothicVR/Scenes/Bootstrap.unity [hunk 6]
```diff
@@ -1064,6 +1064,7 @@ MonoBehaviour:
   VObPerFrame: 75
   moveSpeedPlayerPref: MoveSpeed
   turnSettingPlayerPref: TurnSetting
+  turnAngelPlayerPref: TurnAngelSetting
```
Comment (JaXt0r) [inline Assets/GothicVR/Scenes/Bootstrap.unity:1067]:
```suggestion
  turnAnglePlayerPref: TurnAngleSetting
```

File: Assets/GothicVR/Scripts/UI/TurnAngelController.cs
Comment (JaXt0r) [inline Assets/GothicVR/Scripts/UI/TurnAngelController.cs:1]: Please rename to "TurnAngleController.cs" angel->angle

File: Assets/GothicVR/Scripts/UI/TurnAngelController.cs [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+using UnityEngine.XR.Interaction.Toolkit;
+using GVR.Manager;
+
+public class TurnAngleController : MonoBehaviour
+{
+    public ActionBasedSnapTurnProvider snapturnprovider;
+    public ActionBasedContinuousTurnProvider continuousturnprovider;
+    void Start()
+    {
+        Slider turnangelslider = transform.GetComponent<Slider>();
```
Comment (JaXt0r) [inline Assets/GothicVR/Scripts/UI/TurnAngelController.cs:12]:
```suggestion
        Slider turnAngleSlider = transform.GetComponent<Slider>();
```

**Review by JaXt0r — CHANGES_REQUESTED** [review]
Just a few typos and some ideas left:
1. I found out how to flip the image for back side of the menu. I couldn't find how you apply "back side" but you can easily create the flipped image. https://discussions.unity.com/t/flipping-texture/51754/3
image
Alternative is to have a second mesh which is same pixel dimensions but turned.
2. I remember Frank built, that we can close the menu via >A< again. Is this only in main?
3. You need to merge main into this branch. Let me provide you some help.

**Pull request closed without merging**